packages/ui/src/ui/store/selectors/groups.ts

```diff
@@ -88,15 +88,38 @@ const getGroupsTreeFiltered = createSelector(
 );
 
 const getGroupsTreeFilteredAndExpanded = createSelector(
-    [getGroupsTreeFiltered, getGroupsExpanded],
-    (root, expanded) => {
-        const res = cloneDeep_(root);
+    [getGroupsTree, getGroupsTreeFiltered, getGroupsExpanded, getGroupsNameFilter],
+    (groupTree, groupsTreeFiltered, expandedByUser, groupNameFilter) => {
+        const expandedBySearch: Record<string, boolean> = {};
+
+        const res = cloneDeep_(groupsTreeFiltered);
+
         hammer.treeList.treeForEach(res.children, (node: GroupsTreeNode) => {
-            const {name} = node;
-            if (!expanded[name]) {
+            const isNodeMatchedFilter = groupNameFilter && node.name.includes(groupNameFilter);
+
+            if (isNodeMatchedFilter) {
+                let parentName = node.parent!;
+
+                while (groupTree[parentName]) {
+                    expandedBySearch[parentName] = true;
+                    parentName = groupTree[parentName].parent;
```

Comment:
Our internal version of UI builds with the error:
```
ERROR in ./src/ytsaurus-ui.ui/store/selectors/groups.ts:[32m[1m105:21[22m[39m
TS2322: Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'.
    103 |                 while (groupTree[parentName]) {
    104 |                     expandedBySearch[parentName] = true;
  > 105 |                     parentName = groupTree[parentName].parent;
        |                     ^^^^^^^^^^
    106 |                 }
    107 |             }
    108 |         });
```
Something wrong with build of os-version of UI, it doesn't catch the error, we need to fix it someday.
Please add `!` character at the end of the line:
```ts
parentName = groupTree[parentName].parent!;
```